bin/inx-chronicle/src/api/v2/responses.rs

```diff
@@ -0,0 +1,221 @@
+pub struct MessagesForQueryResponse {
+    pub query: MessagesQuery,
+    #[serde(rename = "maxResults")]
+    pub max_results: usize,
+    pub count: usize,
+    #[serde(rename = "messageIds")]
+    pub message_ids: Vec<Expansion>,
+}
+
+impl_success_response!(MessagesForQueryResponse);
+
+/// Response of GET /api/v2/addresses/<address>/outputs
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct OutputsForQueryResponse {
+    pub query: OutputsQuery,
+    #[serde(rename = "maxResults")]
+    pub max_results: usize,
+    pub count: usize,
+    #[serde(rename = "outputIds")]
+    pub output_ids: Vec<Expansion>,
+}
+
+impl_success_response!(OutputsForQueryResponse);
+
+/// Response of GET /api/v2/outputs/<output_id>
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct OutputResponse {
+    #[serde(rename = "messageId")]
+    pub message_id: String,
+    #[serde(rename = "transactionId")]
+    pub transaction_id: String,
+    #[serde(rename = "outputIndex")]
+    pub output_index: u16,
+    #[serde(rename = "spendingTransaction")]
+    pub spending_transaction: Option<Value>,
+    pub output: Value,
+}
+
+impl_success_response!(OutputResponse);
+
+/// Response of GET /api/v2/transactions/<message_id>
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TransactionResponse {
+    /// The created output's message id
+    #[serde(rename = "messageId")]
+    pub message_id: String,
+    /// The confirmation timestamp
+    #[serde(rename = "milestoneIndex")]
+    pub milestone_index: Option<u32>,
+    /// The output
+    pub outputs: Vec<Value>,
+    /// The inputs, if they exist
+    pub inputs: Vec<Value>,
+}
+
+impl_success_response!(TransactionResponse);
+
+/// Response of GET /api/v2/transactions/ed25519/<address>
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TransactionsResponse {
+    pub transactions: Vec<TransactionResponse>,
+}
+
+impl_success_response!(TransactionsResponse);
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct TransactionHistoryResponse {
+    pub address: String,
+    pub transactions: Vec<Transfer>,
+}
+
+impl_success_response!(TransactionHistoryResponse);
+
+/// Response of GET /api/v2/milestone/<index>
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct MilestoneResponse {
+    #[serde(rename = "index")]
+    pub milestone_index: u32,
+    #[serde(rename = "messageId")]
+    pub message_id: String,
+    pub timestamp: u32,
+}
+
+impl_success_response!(MilestoneResponse);
+
+/// Response of GET /api/v2/analytics/addresses[?start_timestamp=<u32>&end_timestamp=<u32>]
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct AddressAnalyticsResponse {
+    #[serde(rename = "totalAddresses")]
+    pub total_addresses: u64,
+    #[serde(rename = "receivingAddresses")]
+    pub recv_addresses: u64,
+    #[serde(rename = "sendingAddresses")]
+    pub send_addresses: u64,
+}
+
+impl_success_response!(AddressAnalyticsResponse);
```

Comment:
Let's keep the analytics out of `api/v2/` for now. We need to put those routes behind a configuration flag anyway. Everything behind `/api/v2/` should mimic [the TIP](https://editor.swagger.io/?url=https://raw.githubusercontent.com/iotaledger/tips/stardust-api/tips/TIP-0025/core-rest-api.yaml) as much as possible.